src/GenericContainer/tests/GenericContainerTest.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file GenericContainerTest.cpp
+ * @authors Stefano Dafarra
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// Catch2
+#include <catch2/catch.hpp>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Core/TestUtils.h>
+#include <memory>
+#include <vector>
+
+#include <BipedalLocomotionControllers/GenericContainer.h>
+
+using namespace BipedalLocomotionControllers;
+
+TEST_CASE("Generic Container")
+{
+    SECTION("Copy")
+    {
+        iDynTree::VectorDynSize vector(5);
+        iDynTree::getRandomVector(vector);
+        GenericContainer container(iDynTree::make_span(vector));
+
+        std::vector<double> copiedIn;
+        copiedIn.resize(5);
+        GenericContainer containerToBeCopied(iDynTree::make_span(copiedIn));
+
+        containerToBeCopied = container;
+
+        for (long i = 0; i < container.size(); ++i)
+        {
+            REQUIRE(vector[i] == copiedIn[i]);
+        }
+    }
+
+    SECTION("Impossible to resize")
+    {
+        iDynTree::VectorDynSize vector(5);
+        GenericContainer container = makeGenericContainer(vector);
+        REQUIRE_FALSE(container.resize(2));
+
+        iDynTree::VectorFixSize<3> fixedVector;
+        GenericContainer container2 = makeGenericContainer(fixedVector, GenericContainerMode::Resizable);
+        REQUIRE_FALSE(container.resize(2));
+    }
```

Comment:
Correct me if I'm mistaken. `makeGenericContainer` generates a container that is not resizable. If you want to resize it `GenericContainerMode::Resizable` has to be added.  However, if you pass a vector that it is `FixedSize` the `resize` method it will return false

Reply:
Yes, you are right.